pkg/operation/botanist/logging.go

```diff
@@ -163,7 +164,8 @@ func (b *Botanist) destroyShootNodeLogging(ctx context.Context) error {
 
 func (b *Botanist) isShootNodeLoggingEnabled() bool {
 	if b.Shoot != nil && b.Shoot.IsLoggingEnabled() && b.Config != nil &&
-		b.Config.Logging != nil && b.Config.Logging.ShootNodeLogging != nil {
+		b.Config.Logging != nil && gardenlethelper.IsLoggingEnabled(b.Config) &&
+		b.Config.Logging.ShootNodeLogging != nil {
```

Comment:
Maybe here we should consider if the Loki is disabled because without it there is no sense to have `shootnode` logging at all.
```suggestion
		b.Config.Logging.ShootNodeLogging != nil  && 
		b.Config.Logging.Loki != nil &&
		b.Config.Logging.Loki.Enabled != nil &&
		*b.Config.Logging.Loki.Enabled {
```

Reply:
It's not required cause if Loki is not enabled it will stop [here](https://github.com/gardener/gardener/blob/04d2b3f459d198e8db0ab57180ca2fea18e84da9/pkg/operation/botanist/logging.go#L50-L52) only.

Reply:
The link refers to the logic which deploys the Loki in the shoot controlplane.
[This](https://github.com/gardener/gardener/blob/45da551bad81c5a3978cf2186b37284d0399dc50/pkg/operation/botanist/operatingsystemconfig.go#L61) is the logic which install the promtail on the shoot nodes

Reply:
Previously this check was not there, I am not sure was it done intentionally in #3813.

Reply:
This is because it was omitted by mistake. But ok you could simply ignore it. I will fix it in another PR